### ISSUE TITLE
Quarto shiny document app don't show preview on viewer pane

### ISSUE THREAD

**Issue by cderv** [rstudio/rstudio#14039]
### System details

    RStudio Edition : Desktop [Open Source]
    RStudio Version : 2023.12.0.363
    OS Version      : Windows 11 x64 (build 22631)
    R Version       : R version 4.3.1 (2023-06-16 ucrt)

### Steps to reproduce the problem

* Create a new Quarto interactive shiny document
* Run Document in IDE
* Render pane opens 
* But no preview

### Describe the problem in detail

The app does not open in IDE. Render pane shows local host url and in shows ok there

https://github.com/rstudio/rstudio/assets/6791940/a43586f4-1cd5-4c28-9c6a-03e6af1c5530

### Describe the behavior you expected

Apps open in the viewer. I believe it was working before, especially at the time of 

* https://github.com/rstudio/rstudio/issues/12546

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@jonvanausdeln Can you try this out on Windows? It works properly for me on MacOS 14.2 Intel.

**Comment by jonvanausdeln** [comment]
@ronblum , it works as expected for me on 2023.12.0+363. Windows 11, R 4.3.1

<img width="831" alt="image" src="https://github.com/rstudio/rstudio/assets/6433789/13f48db6-c4b8-4077-9628-d46bb586e5b6">

**Comment by cderv** [comment]
Interesting. This is something else in my environment that is causing this then. I'll try to find what does create this for me... 🤔

**Comment by kyleGrealis** [comment]
> Interesting. This is something else in my environment that is causing this then. I'll try to find what does create this for me... 🤔

Has this been updated/corrected? I'm using Ubuntu 22.04, RStudio v.2023.09.1+494, Quarto v.1.4.533 and encountered the same issue today creating a dashboard.

**Comment by cderv** [comment]
Thanks for chiming in @kyleGrealis ! I see I forgot to mention the quarto version I used. It is also Quarto 1.4 

@jonvanausdeln did you try with Quarto pre-release 1.4 ?

**Comment by jthomasmock** [comment]
> I'm using Ubuntu 22.04, RStudio v.2023.09.1+494, Quarto v.1.4.533 and encountered the same issue today creating a dashboard.

Thanks Christophe! Can you confirm if this is a 1.3 vs 1.4 Quarto issue or something else in your environment perhaps?

**Comment by cderv** [comment]
This is interesting : This happens with Quarto 1.4 and not 1.3 

<details>
<summary>1.3 output</summary>

````
==> (function(input, ...) { invisible(system(paste0('quarto serve "', input, '" ', ''))) })('C:/Users/chris/Documents/test.qmd', shiny_args = list(launch.browser = FALSE), auto_reload = FALSE, dir = 'C:/Users/chris/Documents',  render_args = list(encoding = 'UTF-8', output_format = 'html'));
++ Activating rlang global_entrace

++ Activating rlang global_entrace

Le chargement a nécessité le package : shiny


processing file: test.qmd
                                                                                                            
output file: test.knit.md

pandoc 
  to: html
  output-file: test.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png
  
metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: Untitled
  server: shiny
  
++ Activating rlang global_entrace

Le chargement a nécessité le package : shiny

Listening on http://127.0.0.1:7822
````

</details>

<details>
<summary>1.4 output</summary>

==> (function(input, ...) { invisible(system(paste0('quarto serve "', input, '" ', ''))) })('C:/Users/chris/Documents/test.qmd', shiny_args = list(launch.browser = FALSE), auto_reload = FALSE, dir = 'C:/Users/chris/Documents',  render_args = list(encoding = 'UTF-8', output_format = 'html'));
++ Activating rlang global_entrace

++ Activating rlang global_entrace

Le chargement a nécessité le package : shiny


processing file: test.qmd
                                                                                                            
output file: test.knit.md

pandoc 
  to: html
  output-file: test.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png
  
metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: Untitled
  server:
    type: shiny
  
++ Activating rlang global_entrace

Le chargement a nécessité le package : shiny
Browse at http://localhost:7328/


</details>

In short: 

* In 1.4 we now log `Browse at http://localhost:7328/` 
* In 1.3 we logged `Listening on http://127.0.0.1:7822`

And currently IDE is reacting to the latter only 
https://github.com/rstudio/rstudio/blob/826d4f2da0fc4b85da265d91929cdfea3f1c4115/src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp#L715-L720

So either IDE needs to adapt, but it may be the other way around and Quarto should not change the way it outputs. I need to understand why because I do think we also rely on this internally 🤔

**Comment by kevinushey** [comment]
It's probably easier to just adapt the RStudio IDE here? It should be a quick fix nonetheless.

**Comment by kevinushey** [comment]
Hmm, I just tried this locally and I don't see any prefix at all:

```
> system("quarto --version")
1.4.537
```

<img width="351" alt="Screenshot 2024-01-09 at 10 58 14 AM" src="https://github.com/rstudio/rstudio/assets/1976582/4e573382-9588-4eec-817b-911243eb9194">

**Comment by kevinushey** [comment]
I also suspect the coloring / use of ANSI escapes might confuse our regex parser here.

**Comment by cderv** [comment]
> Hmm, I just tried this locally and I don't see any prefix at all

puzzling... 🤔 

I get this : 
![image](https://github.com/rstudio/rstudio/assets/6791940/595bc4cb-f180-4400-90be-21ec49de22d0)


I am on Window though, and no color for me (🤷) on 2024.04.0 Build 192

Are you on RStudio server ? It seems we do not print there for some reason: 
https://github.com/quarto-dev/quarto-cli/blob/9f6968358fb00034c08cc613294c906a184a853a/src/core/previewurl.ts#L87-L95

I don't know well that part of the code yet. so looking as we try to understand.

**Comment by cderv** [comment]
Proposed fix in Quarto by @dragonstyle 
* https://github.com/quarto-dev/quarto-cli/pull/8191

**Comment by kevinushey** [comment]
To confirm, I was indeed working with RStudio Server.

**Comment by kevinushey** [comment]
I've also committed a fix to RStudio here: https://github.com/rstudio/rstudio/commit/fb1838bc0593abadf4ccf1d55a8ad31e600c2391

Once the next daily builds are live, would you be able to test?

**Comment by cderv** [comment]
@kevinushey working ok with 2024.04.0-daily+200 ! 

Thanks

**Comment by cderv** [comment]
We merged the PR in 1.4 which emits now `Listening on` as support by RStudio. This will ensure it is working with current and previous RStudio IDE version 

Available in [v1.4.539](https://github.com/quarto-dev/quarto-cli/releases/tag/v1.4.539)

**Comment by kevinushey** [comment]
@cderv: would you be willing to test with a recent daily build of RStudio and let us know if the issue persists?

**Comment by cderv** [comment]
Works ok when I test in 
```
RStudio 2023.12.0+369 "Ocean Storm" Release (33206f75bd14d07d84753f965eaa24756eda97b7, 2023-12-17) for windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.12.0+369 Chrome/116.0.5845.190 Electron/26.2.4 Safari/537.36
```

with Quarto 1.5 dev version 

So all seems good.

**Comment by kevinushey** [comment]
Awesome news -- thank you!
